sticht/rollbacks/sources/splunk.py

```diff
@@ -116,52 +119,46 @@ def process_result(self, result: Optional[List[Dict[Any, Any]]], earliest_timest
         number of results from a query against configured thresholds to determine
         whether or not to rollback or not
         """
-        if earliest_timestamp_seconds != 'now':
-            if earliest_timestamp_seconds < self.start_timestamp_seconds:
-                self.bad_before_mark = self.is_window_bad(result)
-        else:
+        if earliest_timestamp_seconds != 'now' and earliest_timestamp_seconds > self.start_timestamp_seconds:
             self.bad_after_mark = self.is_window_bad(result)
+        else:
+            self.bad_before_mark = self.is_window_bad(result)
 
         old_failing = self.failing
+
         self.failing = self.bad_after_mark and not self.bad_before_mark
+
         if self.failing == (not old_failing):
             self.on_failure_callback(self)
 
     def is_window_bad(self, result) -> bool:
         if self._query_type == 'results':
-            if self._lower_bound and len(result) < self._lower_bound:
-                return True
-            elif self._upper_bound and len(result) > self._upper_bound:
-                return True
-        else:
-            result_num = next(iter(result[0].values()))
-            if self._lower_bound and result_num < self._lower_bound:
-                return True
-            elif self._upper_bound and result_num > self._upper_bound:
-                return True
-        return False
+            if self._lower_bound and self._upper_bound:
+                if len(result) > self._lower_bound and len(result) < self._upper_bound:
+                    return True
+            elif self._lower_bound:
+                if len(result) > self._lower_bound:
+                    return True
+            elif self._upper_bound:
+                if len(result) < self._upper_bound:
+                    return True
+        return True
 
     @classmethod
     def from_config(  # type: ignore[override]
         # we don't care about violating LSP - we just want to follow a given interface
         cls: Type['SplunkMetricWatcher'],
         config: SplunkRule,
-        check_interval_s: Optional[float],
         on_failure_callback: Callable[['MetricWatcher'], None],
         auth_callback: Callable[[], SplunkAuth],
     ) -> 'SplunkMetricWatcher':
-        if check_interval_s is not None:
-            log.warning(
-                f'Ignoring check_interval_s of {check_interval_s}'
-                + f'and using default of {DEFAULT_SPLUNK_CHECK_INTERVAL_S}',
-            )
-
         return cls(
             config['label'],
             config['query'],
             config['query_type'],
             on_failure_callback=on_failure_callback,
             auth_callback=auth_callback,
+            check_interval_s=config['check_interval_s'] if 'check_interval_s' in config else DEFAULT_CHECK_INTERVAL_S,
```

Comment:
do we restrict people from being able to specify something ilke `check_interval_s: null` so we're guaranteed this is a real number? `'check_interval_s'` would be in `config` in this case, but then it would accept the value as None here.

Reply:
We only allow integers for this key.
https://github.com/Yelp/paasta/blob/master/paasta_tools/cli/schemas/rollback_schema.json#L155